docs/source/concepts/bento.rst

```diff
@@ -763,30 +763,216 @@ Docker Template (Danger Zone)
 """""""""""""""""""""""""""""
 
 The :code:`docker_template` field gives user the full control over how the
-:code:`Dockerfile` was generated in a Bento.
+:code:`Dockerfile` was generated in a Bento. Users can use this field to
+customize Bento's Dockerfile instruction set to suits their needs.
 
-Documenting this option is working-in-progress, see :issue:`2497`
+Let say that you want to isolate the installation of a custom python library from a build stage to the 
+Bento utilizing `dockerfile's network modes <https://github.com/moby/buildkit/blob/master/frontend/dockerfile/docs/syntax.md#network-modes-run---networknonehostdefault>`_. You can use the :code:`docker_template` field to do this.
 
+First, write a :code:`Dockerfile.template` file anywhere that fits your needs.
+This template file is a mixed between a :code:`Dockerfile` and a :code:`Jinja` template file:
 
+.. code-block:: dockerfile
+
+   {% extends bento__dockerfile %}
+   {% block SETUP_BENTO_BASE_IMAGE %}
+   FROM --platform=$BUILDPLATFORM python:3.7-slim as buildstage
+   RUN mkdir /tmp/mypackage
+
+   WORKDIR /tmp/mypackage/
+   COPY mypackage .
+   RUN python setup.py sdist && mv dist/mypackage-0.0.1.tar.gz mypackage.tar.gz
+
+   {{ super() }}
+   {% endblock %}
+   {% block SETUP_BENTO_COMPONENTS %}
+   {{ super() }}
+   COPY --from=buildstage mypackage.tar.gz /tmp/wheels/
+   RUN --network=none pip install --find-links /tmp/wheels mypackage
+   {% endblock %}
+
+.. note::
+
+   :bdg-warning:`Warning:` Remember to change your :code:`mypackage-0.0.1.tar.gz` to the correct tarfile name.
+
+.. note::
+
+   The template file can have extension :code:`.jinja`, :code:`.j2`, or any
+   extensions that Jinja2 can understand.
+
+Then add the path to the given template file to the :code:`docker_template` field in the :code: `bentofile.yaml`:
+
+.. code:: yaml
+
+    docker:
+        docker_template: "~/workspace/Dockerfile.template"
+
+Now to see the result generated Dockerfile, do the following:
+
+.. code-block:: bash
+
+   bentoml build && cat $(bentoml get <bento> -o path)/env/docker/Dockerfile
+
+.. dropdown:: Results should looks something like this: 
```

Comment:
remove the full dockerfile generated since it's hard for us to keep updating it in docs. just let user to confirm the line is in the generated dockerfile

Reply:
noted